File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    result = []
    
    s1 = [1, 2, 3, 4, 5]
    s2 = [2, 1, 2, 3, 2, 4, 2, 5]
    s3 = [3, 3, 1, 1, 2, 2, 4, 4 , 5, 5]
    
    s1_cnt = 0
    s2_cnt = 0
    s3_cnt = 0
    for i in range(len(answers)):
        if s1[i%len(s1)] == answers[i]:
            s1_cnt += 1
        if s2[i%len(s2)] == answers[i]:
            s2_cnt += 1
        if s3[i%len(s3)] == answers[i]:
            s3_cnt += 1
    
    cnt = [s1_cnt, s2_cnt, s3_cnt]
    for i in range(3):
        if cnt[i] == max(cnt):
            result.append(i+1)
    
    return result

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const result = [];
    let c = [0, 0, 0];
    const a1 = [1, 2, 3, 4, 5];
    const a2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const a3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    for (let i = 0; i < answers.length; i++) {
        if (answers[i] === a1[i % a1.length]) c[0]++;
        if (answers[i] === a2[i % a2.length]) c[1]++;
        if (answers[i] === a3[i % a3.length]) c[2]++;
    }
    const max = Math.max(...c);
    for (let i = 0; i < 3; i++) {
        if (c[i] === max) result.push(i + 1);
    }
    return result;
}